### ISSUE TITLE
Docker --log-level implemented, but not documented

### ISSUE THREAD

**Issue by RoelVdP** [docker/docker#10034]
$ docker --help 2>&1 | grep log-level
  -l, --log-level="info"                 Set the logging level

This was implemented, likely as a result of https://github.com/docker/docker/issues/5198, yet it is not documented anywhere on the Docker website it seems. Also, it is not documented in man docker, and docker --help only lists the existence of the option, but not it's usage details.

**Comment by duglin** [comment]
I do see it here: https://docs.docker.com/reference/commandline/cli/
and in man here: https://github.com/docker/docker/blob/master/docs/man/docker.1.md
where else did you have in mind?

**Comment by RoelVdP** [comment]
CLI, latest iteration, only lists "info" level (to see all available levels would be great);

$ docker --version
Docker version 1.4.1, build 5bc2ff8
[roel@localhost percona-qa]$ docker --help 2>&1 | grep -B2 -A2 "\-\-log"
  --ip-masq=true                             Enable IP masquerading for bridge's IP range
  --iptables=true                            Enable Docker's addition of iptables rules
  -l, --log-level="info"                     Set the logging level           <---- please list possible options here
  --label=[]                                 Set key=value labels to the daemon (displayed in `docker info`)
  --mtu=0                                    Set the containers network MTU

https://docs.docker.com/reference/commandline/cli/ has the same info: i.e. no levels listed

https://github.com/docker/docker/blob/master/docs/man/docker.1.md :
Interesting, but this does not show yet on for example latest Centos-provided Docker Release (1.3.x iirc) - even when the binary (using the wget method described in the manual) has been updated - because the corresponding man page does not get updated accordingly.

Maybe the answer for this latter issue is to give an additional wget in the manual (at https://docs.docker.com/installation/binaries/#upgrades for example) to fetch the man page as well (and if so, what URL to use etc.)

**Comment by duglin** [comment]
Opened PR #10124 to cover this.
I didn't address the last point of your comment because it seems like a CoreOS issue and I'm not sure I would want to document a way to put your docs in a 1/2 upgraded/inconsistent state.  But if the doc maintainers want something in that space I can work on it.

**Comment by RoelVdP** [comment]
@duglin Thank you for the patch for the CLI, great. 

The 1/2 upgraded/inconsistent state is a recommended procedure by Docker (and it's a nice/handy one at that - especially for distro's that have the package included, but an older version), so may as well add the man page update also. 

Side question; what and how many other files are there besides the main binary and the man page?

**Comment by RoelVdP** [comment]
https://docs.docker.com/reference/commandline/cli/ page update - will this happen automatically once the CLI help refreshes?

**Comment by RoelVdP** [comment]
Oh, btw, no CoreOS, it's Centos.

**Comment by duglin** [comment]
@RoelVdP I assume that .../cli/ page will be automatically updated after the new rev of Docker.
Where is the page that talks about the partial upgrade stuff?
@SvenDowideit thoughts on the partial upgrade?

**Comment by RoelVdP** [comment]
https://docs.docker.com/installation/binaries/#upgrades

**Comment by duglin** [comment]
yea I saw that but that's doesn't talk about partial upgrades, just a manual one - which I assume upgrades all of it.

**Comment by RoelVdP** [comment]
It seems it does;

```
Get the docker binary:
$ wget https://get.docker.com/builds/Linux/x86_64/docker-latest -O docker
$ chmod +x docker
```

This just upgrades the binary, not the manual?

**Comment by duglin** [comment]
Right so adding text to upgrade the man pages, docs, etc... is something I'd want @SvenDowideit 's input on - and perhaps do it under a new issue/PR because I think that's almost a different issue.

**Comment by RoelVdP** [comment]
@SvenDowideit - thoughts?

**Comment by BretFisher** [comment]
TL;DR answer: `--log-level` on dockerd can be one of `debug, info, warn, error, fatal`

**Comment by sab24** [comment]
Indeed, who closed this issue? This is the only place I can find that --log-level can have the options of debug, info, warn error and fatal. It is not in the documentation.

**Comment by thaJeztah** [comment]
@sab24 If you look two comments up, you can see that documentation changes were added last year in https://github.com/docker/docker/pull/10124, and therefore this issue was closed.

see https://github.com/docker/docker/pull/26894#issue-179089873 for an explanation why that help text was not there anymore. PR https://github.com/docker/docker/pull/26894 was created to fix this again, and and will be in the next release. Documentation changes will be published together with the release but can be read here in this repository; https://github.com/docker/docker/blob/master/docs/reference/commandline/dockerd.md

**Comment by sab24** [comment]
thaJeztah, Thanks for clarifying. There are no man pages installed on MacOS, so it is difficult to find this information. Good that it will be fixed.
